assets/src/js/menu_editor.js

```diff
@@ -0,0 +1,349 @@
+      'li.menu-item:not(.sortable-placeholder)'
+    );
+
+    // Pull back menu items to the limited depth
+    items.forEach((item) => {
+      normalizeDepth(item, conf.maxDepth);
+    });
+
+    // Check root level
+    const rootItems = document.querySelectorAll('.menu-item-depth-0:not(.sortable-placeholder)');
+    const rootConf = getDepthConf(conf, 0);
+    if (rootItems.length > rootConf.maxItems) {
+      markItemsOverflow(errors, rootItems, rootConf.maxItems, __('Main menu', 'planet4-master-theme-backend'));
+    }
+
+    // Check each item for submenus and titles too long
+    items.forEach((item) => {
+      const depth = getItemDepth(item);
+      const children = getItemChildren(item);
+      const depthConf = getDepthConf(conf, depth);
+
+      const itemDiv = item.querySelector('.menu-item-title');
+      const title = itemDiv ? itemDiv.textContent : '';
+
+      // Too many children for the given item
+      const subLevelConf = getDepthConf(conf, depth+1);
+      if ( children && children.length > subLevelConf.maxItems ) {
+        markItemsOverflow(
+          errors,
+          children,
+          subLevelConf.maxItems,
+          sprintf(__('Sub menu "%s"', 'planet4-master-theme-backend'), title)
+        );
+      }
+
+      // Item title is too long
+      const titleCharsCount = [...title].length; // Count chars instead of code units
+      if (titleCharsCount > depthConf.maxChars) {
+        errors.push({
+          target: item.id,
+          message: sprintf(
+            __(
+              'This item label "%1$s" should not be longer than %2$s characters (currently %3$s).',
+              'planet4-master-theme-backend'
+            ),
+            title, depthConf.maxChars, titleCharsCount
+          )
+        });
+      }
+    });
+
+    // Display error messages and mark items in error
+    displayErrors(errors);
+  };
+
+  /**
+   * Get first checked location
+   *
+   * @return string The current location slug
+   */
+  const getCurrentLocation = () => {
+    return locations.map(location => {
+      return document.querySelector(`input[name="menu-locations[${location}]"]`).checked
+        ? location : null;
+    })[0] || null;
+  };
+
+  /**
+   * Gets the current conf.
+   *
+   * @return Object The current conf
+   */
+  const getCurrentConf = () => {
+    const location = getCurrentLocation();
+    const conf = p4_menu_config[location] || null;
+    if (!conf) {
+      return;
+    }
+
+    return {...{ maxItems: 5, maxChars: 18, maxDepth: 1, }, ...conf};
+  };
+
+  /**
+   * Get item children
+   *
+   * @param NodeElement item The item
+   * @return NodeElement[]|null Children list of this item
+   */
+  const getItemChildren = (item) => {
+    const depth = getItemDepth(item);
+    const itemsAfter = menuContent.querySelectorAll(
+      `#${item.id} ~ li.menu-item:not(.sortable-placeholder)`
+    );
+    if (itemsAfter.length <= 0 ) {
+      return null;
+    }
+
+    let children = [];
```

Comment:
I think I can't break out of a `.filter()`, I use a `for` loop here because I want to stop when I reach a different level - so trying to get the sub items of an item only on the specific sublevel
```
- Item A
- Item B
  - Item B-1
  - Item B-2
    - Item B-2-1
  - Item B-3
- Item C
```
So here for `getItemChildren(Item B)`, only the B-\* items, not A/C or B-\*-\*. And now that I write this I realize the issue in my code that will stop too early in some cases (it's not visible because of the depth limit currently used) :grimacing: And also missing a reset of the globalMaxDepth, fixing that :ok_hand: